### PR TITLE
feat: utility methods for fetching tweets using a query

### PR DESCRIPTION
**Pull request by Owen3H**
- implement `getTweetWhere`. Gets the first tweet matching the query, or null if no match.
- implement `getTweetsWhere`. An array of all tweets matching the query (or empty).
- support passing optional `max` variable to `getLatestTweet`.

Tests and documentation included for both new methods.

### CODE REVIEW

File: src/tweets.ts
Comment (karashiiro) [inline src/tweets.ts:179]: I think there's something like `Iterable<T>` for this, I don't remember which TS targets have that, though.

**Review by karashiiro — CHANGES_REQUESTED** [review]
Could we have this take a callback instead of a match object? That would make this flexible enough to enable things like regex matching as well. Effectively, that would make this an inline `filter`. I can see a use case for dropping in ts-pattern on the consumer side to do per-field functional matching, for example.

Better yet, the callback could return `boolean | Promise<boolean>` to allow for more niche setups as well (maybe a branch into async logic under certain conditions).

File: src/tweets.ts
Comment (karashiiro) [reply src/tweets.ts:179]: Should be [`AsyncIterator<T>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncIterator), which I think supports sync iteration as well?

**Comment by Owen3H** [comment]
I gave your suggestions a shot, some things to be noted:

- I believe `AsyncIterable<Tweet>` should suffice, let me know if you meant something more generic.
- Im not sure how you would like the docs updated so I best leave that to you ;)
- Since exporting the same method while overloading, I've combined them into the same function with a simple isCallback so both an object or callback can be accepted.

**Comment by karashiiro** [comment]
Looks good, just noted a minor issue when invoking the callback. I'll revise the docs for the query parameter after merging.

**Pull request merged**
